spec/unit/indirector/catalog/compiler_spec.rb

```diff
@@ -493,6 +493,34 @@ def stubs_directory_metadata(relative_path)
       expect(catalog.recursive_metadata).to be_empty
     end
 
+    it "uses file parameters which match the true file type defaults" do
+      catalog = compile_to_catalog(<<-MANIFEST, node)
+        file { '#{path}':
+          ensure => file,
+          source => '#{source}'
+        }
+      MANIFEST
+
+      if Puppet::Util::Platform.windows?
+        default_file = Puppet::Type.type(:file).new(:name => 'C:\defaults')
+      else
+        default_file = Puppet::Type.type(:file).new(:name => '/defaults')
+      end
+
+      metadata = stubs_file_metadata(checksum_type, checksum_value, 'modules/mymodule/files/config_file.txt')
+
+      options = {
+        :environment => catalog.environment_instance,
+        :links => default_file[:links],
```

Comment:
This could also be done in a few existing test cases, though that may muddy what they're testing and why they fail.